### ISSUE TITLE
Relation and custom id not working

### ISSUE THREAD

**Issue by nininhocovre** [Stuart-campbell/RushOrm#68]
I have the table relationship like below:

    public class Talk extends RushObject {
        private Long id;
        private String title;
        private Room room;
        @RushList(classType = Speaker.class)
        private List<Speaker> speakers;
    }

    public class Room extends RushObject {
        private Long id;
        private String name;
    }

    public class Speaker extends RushObject {
        private Long id;
        private String name;
    }

So all the items have an id that comes from a server, so I would like to keep them.
The problem is that a Talk can have the same Room or same Speaker as another Talk, so when I save the Talk, it duplicates the Room and Speaker.

I tried using registerObjectWithId() on the objects and also overriding the getId method from RushObject but that wouldn't work because in the relation some tables would have my id and others would be saved with the generated id, so loading the Talk from database would have no Room or Speaker within because the ids of them are different and the join would fail.

If I don't set the id the relationship works, but my id would not be unique so every time I get the data from the server the data gets saved again and duplicated.

To sum it up, I would like to know if there is a way to make a column unique or set the id of the object in a way that relationship works.

**Comment by Stuart-campbell** [comment]
Hi,

I think the problem might just have been overriding the getId method. This should work.

    public class Talk extends RushObject {
        private Long id;
        private String title;
        private Room room;
        @RushList(classType = Speaker.class)
        private List<Speaker> speakers;

        public void save() {
            RushCore.getInstance(). registerObjectWithId(this, id);
            RushCore.getInstance(). registerObjectWithId(room, room.getServerId());
            for(Speaker speaker : speakers) {
                RushCore.getInstance(). registerObjectWithId(speaker, speaker.getServerId());
            }
            super.save();
        }
    }

Or if you server is sending you json that matches the above object structure you could try this.

    JSONObject json = new JSONObject();
    JSONArray array = new JSONArray();
    array.put(new JSONObject("---json representing Talk as root object-- -"));
    json.put(Talk.class.getSimpleName(), array);
    List<Talk> talks = RushCore.getInstance(). deserialize(json, "id", Talk.class);
    Talk talk = talks.get(0);
    talk.save();

Thanks

**Comment by nininhocovre** [comment]
Thanks, that worked
